tests/scripts/all.sh

```diff
@@ -65,6 +66,8 @@ General options:
   -f|--force            Force the tests to overwrite any modified files.
   -k|--keep-going       Run all tests and report errors at the end.
   -m|--memory           Additional optional memory tests.
+     --armcc            Run ARM Compiler builds (on by default).
```

Comment:
@mpg I'd added a positive option for `--yotta` as well. I prefer to have + and - options when practical, so you can run `all.sh --yotta` and be sure to get the yotta build even if we decide to remove it in the future.

It is inconsistent in that the older options have no inverse. Should I add them? I don't want to put too much effort into this script, I'd like to replace it by something more robust (in Python, I guess) in the medium term.

Reply:
I think it's OK to have positive and negative options only for things that enable/disable sets of tests, ie currently yotta and armcc, and only have one-way options for the rest.

Also, I'm a bit curious as to what problem(s) you see with this script that you think could be improved by a python rewrite?

Reply:
What I'd really like is to have the option of running any subset of the tests. CI would run all tests (except perhaps ones that we'd declare as experimental) but it's useful to run a subset locally. And perhaps it would run only a subset on some platforms. I'd also like more robust cleanup and reporting.